pluggy/callers.py

```diff
@@ -202,3 +202,59 @@ def _multicall(hook_impls, caller_kwargs, specopts={}, hook=None):
                 pass
 
         return outcome.get_result()
+
+
+def _itercall(hook_impls, caller_kwargs, specopts={}, hook=None):
+    """Execute a calls into multiple python functions/methods and yield
+    the result(s) lazily.
+
+    ``caller_kwargs`` comes from _HookCaller.__call__().
+    """
+    __tracebackhide__ = True
+    specopts = hook.spec_opts if hook else specopts
+    results = []
+    firstresult = specopts.get("firstresult")
+    excinfo = None
+    try:  # run impl and wrapper setup functions in a loop
+        teardowns = []
+        try:
+            for hook_impl in reversed(hook_impls):
+                try:
+                    args = [caller_kwargs[argname] for argname in hook_impl.argnames]
+                except KeyError:
+                    for argname in hook_impl.argnames:
+                        if argname not in caller_kwargs:
+                            raise HookCallError(
+                                "hook call must provide argument %r" % (argname,))
+
+                if hook_impl.hookwrapper:
+                    try:
+                        gen = hook_impl.function(*args)
+                        next(gen)   # first yield
+                        teardowns.append(gen)
+                    except StopIteration:
+                        _raise_wrapfail(gen, "did not yield")
+                else:
+                    res = hook_impl.function(*args)
+                    if res is not None:
+                        results.append(res)
+                        yield res
```

Comment:
This is literally the only thing added.

Reply:
now that bit feeels really fishy because this is a massive mixup of concepts (sometimes minimal changes are the absolute antithesis of responsible software development)

Reply:
uppon closer inspection tihs mixes iteration with result returning as object, instinctuvely this cant possibly be conceptually sound

Reply:
agreed I'm just pushing out a quick version.
Obviously the final cut should get rid of anything from the original function which isn't necessary in the generator.

Reply:
> uppon closer inspection tihs mixes iteration with result returning as object, instinctuvely this cant possibly be conceptually sound

I'm not sure what you mean, the implementation is very similar to the existing one except for the `yield` part and removing the `return` at the end as @tgoodlet commented.